tests/functional/metrics/required_dependencies/test_metrics_cloudwatch_emf.py

```diff
@@ -443,6 +446,38 @@ def test_schema_validation_no_namespace(metric, dimension):
             my_metric.add_dimension(**dimension)
 
 
+def test_schema_validation_empty_metric_name(metric, dimension, namespace):
```

Comment:
Hey @sdangol all these 2 tests are working as expected, but codecov is complaining because we need to test the `EMFMetricProvider`, so, can you please make a small change and create a new instance of this provider?

Something like this:

`my_metrics = AmazonCloudWatchEMFProvider(namespace=namespace)`

Reference: https://github.com/aws-powertools/powertools-lambda-python/pull/7026/files#diff-53fb0e6eea8f9e23cdca06f0c22375183de0f1ddb845122e6fd73f25c9caa8f3R45